### PR TITLE
i18n(zh-cn): translate tutorial

### PR DESCRIPTION
**Pull request by luoingly**
#### What kind of changes does this PR include?

- Translated content

#### Description

- Chinese translation of the tutorial introduction
- The continuation of another PR: https://github.com/withastro/docs/pull/2704

考虑到原译者目前可能没有时间和精力去继续这项翻译工作，我直接接手了翻译进度并且完成了剩余部分的翻译。为了不打扰原译者，我选择来直接另开一个 PR。当然如果任何人对此有其他想法，可以直接提出来。

Considering that the original translator @KurumiFrost may currently lack the time and energy to continue with this translation work, I have taken over the translation progress directly and completed the remaining part of the translation. In order to not disturb the original translator, I have chosen to open a new PR directly. Of course, if anyone has other ideas about this, they can directly propose them.

我刚刚完成基本的翻译就来开这个 PR 了，尽管我也认为它需要进一步修改，所以刚开始一段时间我可能会把这个 PR 设置为草稿（Draft）。这么早开 PR 的一个原因，就是为了让更多的译者能来审阅并提出修改意见。当我觉得这个翻译已经很完善的时候，我大概就会删除这个草案标记。

I just completed the basic translation and came to open this PR, although I also believe that it needs further modifications, so I might set this PR as a draft for a while at the beginning. One reason for opening the PR this early is to allow more translators to review it and provide suggestions for modifications. When I feel that the translation is already quite refined, I will probably remove the draft mark.

**对于其他译者和贡献者：** 如果你有任何修改意见，可以直接在这个 PR 中提出，也可以修改并提交后去[我的仓库](https://github.com/luoingly/astro-docs/tree/patch-4)开一个 PR。

**For other translators and contributors:** If you have any suggestions for modifications, you can directly propose them in this PR or make modifications and submit a PR in [my repository](https://github.com/luoingly/astro-docs/tree/patch-4).

### CODE REVIEW

**Comment by liruifengv** [comment]
Very large workload. I will help review this pr.

File: src/content/docs/zh-cn/tutorial/1-setup/5.mdx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+<Checklist>
+- [ ] 我可以在线查看我更新后的网站。
+- [ ] 我准备好去编写代码了！
+</Checklist>
+</Box>
+
+### 相关资源
+
+- <p>[在 Netlify 上进行部署的详细指南](https://www.netlify.com/blog/2016/09/29/a-step-by-step-guide-deploying-on- netlify/) <Badge>外部链接</Badge></p>
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/1-setup/5.mdx:79]:
```suggestion
- <p>[在 Netlify 上进行部署的详细指南](https://www.netlify.com/blog/2016/09/29/a-step-by-step-guide-deploying-on-netlify/) <Badge>外部链接</Badge></p>
```

File: src/content/docs/zh-cn/tutorial/2-pages/index.mdx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+## 我们现在在哪？
+
+你可以在 [GitHub](https://github.com/withastro/blog-tutorial-demo/tree/unit-2/start) 或者 [StackBlitz](https://stackblitz.com/github/withastro/blog-tutorial-demo/tree/unit-2/start?file=src/pages/index.astro) 上找到这个教程当前阶段的代码。
+
+检查你的代码是否与此示例匹配。或者如果你刚刚加入我们，请在 StackBlitz 上 Fork，并从这里开始在浏览器中进行编程。
+
+## 本单元目标
+
+在本单元中，你将
+- 使用 .astro 语法创建你的第一个 Astro 页面
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/index.mdx:24]:
```suggestion
- 使用 `.astro` 语法创建你的第一个 Astro 页面
```

File: src/content/docs/zh-cn/tutorial/2-pages/index.mdx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+- 使用 Markdown (`.md`) 文件编写博客文章
+- 使用 `<style>` 设置单个页面的样式
+- 应用跨页面应用全局样式
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/index.mdx:27]:
```suggestion
- 跨页面应用全局样式
```

File: src/content/docs/zh-cn/tutorial/2-pages/1.mdx [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  - 在你的网站上创建两个新页面：关于和博客
+  - 为你的页面添加导航链接
+  - 将你的网站的更新版本部署并部署到线上
+</PreCheck>
+
+## 创建一个新的 `.astro` 文件
+
+1. 在你的代码编辑器的文件窗口中，导航到`src/pages/`文件夹，你会看到现有的`index.astro`文件。
+
+2. 在同一文件夹中，创建一个名为`about.astro'的新文件。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/1.mdx:25]:
```suggestion
2. 在同一文件夹中，创建一个名为`about.astro`的新文件。
```

File: src/content/docs/zh-cn/tutorial/2-pages/2.mdx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+现在你已经使用 `.astro` 文件构建了页面，让我们使用 `.md` 文件编写一些博客文章！
+
+<PreCheck>
+  - 创建一个新文件夹并写一篇新文章
+  - 编写一些 Markdown 内容
+  - 将你的博客文章链接到你的博客页面
+</PreCheck>
+
+## 创建你的第一个 .md 文件
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/2.mdx:24]:
```suggestion
## 创建你的第一个 `.md` 文件
```

File: src/content/docs/zh-cn/tutorial/2-pages/2.mdx [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+## 编写 Markdown 内容
+
+1. 将以下代码复制或键入到 `post-1.md` 中
+
+    ```markdown title="src/pages/posts/post-1.md"
+    ---
+    title: '我的第一篇博客文章'
+    pubDate: 2022-07-01
+    description: '这是我 Astro 博客的第一篇文章。'
+    author: 'Astro 学者'
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/2.mdx:45]:
```suggestion
    author: 'Astro 学习者'
```

File: src/content/docs/zh-cn/tutorial/2-pages/2.mdx [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    image:
+        url: 'https://astro.build/assets/blog/astro-1-release-update/cover.jpeg'
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/2.mdx:47]:
```suggestion
        url: 'https://docs.astro.build/assets/full-logo-light.png'
```

File: src/content/docs/zh-cn/tutorial/2-pages/2.mdx [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+    ```markdown title="src/pages/posts/post-1.md"
+    ---
+    title: '我的第一篇博客文章'
+    pubDate: 2022-07-01
+    description: '这是我 Astro 博客的第一篇文章。'
+    author: 'Astro 学者'
+    image:
+        url: 'https://astro.build/assets/blog/astro-1-release-update/cover.jpeg' 
+        alt: 'The Astro logo with the word One.'
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/2.mdx:48]:
```suggestion
        alt: 'The full Astro logo.'
```

File: src/content/docs/zh-cn/tutorial/2-pages/2.mdx [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+
+2. 现在，在 `src/pages/posts/` 中再添加两个文件：`post-2.md` 和 `post-3.md`。这里有一些示例代码，你可以将其复制并粘贴到你的文件中，或者，你可以编写自己的代码！
+
+    ```md title="src/pages/posts/post-3.md"
+    ---
+    title: 我的第三篇博客文章
+    author: Astro 学者
+    description: "学习了一些 Astro，我根本停不下来！"
+    image: 
+        url: "https://astro.build/assets/blog/astro-showcase/astro-showcase-screenshot.jpg"
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/2.mdx:113]:
```suggestion
        url: "https://docs.astro.build/assets/arc.webp"
```

File: src/content/docs/zh-cn/tutorial/2-pages/2.mdx [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+        <ul>
+          <li><a href="/posts/post-1/">文章1</a></li>
+        </ul>
+      </body>
+    </html>
+    ```
+
+2. 现在，在 `src/pages/posts/` 中再添加两个文件：`post-2.md` 和 `post-3.md`。这里有一些示例代码，你可以将其复制并粘贴到你的文件中，或者，你可以编写自己的代码！
+
+    ```md title="src/pages/posts/post-3.md"
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/2.mdx:107]: The code snippet for `post-2` is missing here.

File: src/content/docs/zh-cn/tutorial/2-pages/2.mdx [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+- [ ] 我可以在 `src/pages/` 中为我的博客文章创建一个新文件夹。
+- [ ] 我可以创建一个新的 Markdown (`.md`) 博客文章文件。
+- [ ] 我知道了 Markdown 是另一种语言，它可以像 Astro 一样在浏览器中生成 HTML。
+</Checklist>
+</Box>
+
+### 相关资源
+
+- [Markdown 指南中的 Markdown 备忘单](https://www.markdownguide.org/cheat-sheet/) <Badge>外部链接</Badge>
+- [什么是浏览器开发者工具？- MDN](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_are_browser_developer_tools) <Badge>外部链接</Badge>
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/2.mdx:178]:
```suggestion
- [什么是浏览器开发者工具？- MDN](https://developer.mozilla.org/zh-CN/docs/Learn/Common_questions/What_are_browser_developer_tools) <Badge>外部链接</Badge>
```
MDN link can use zh-CN

File: src/content/docs/zh-cn/tutorial/2-pages/3.mdx [hunk 12]
```diff
@@ -0,0 +1,279 @@
...
+        <h2>……和我的新 Astro 网站！</h2>
+
+        <p>我正在学习 Astro 的入门教程。这是我网站上的第二个页面，也是我自己建立的第一页面！</p>
+
+        <p>随着我完成更多教程，该站点将更新，所以请继续查看我的旅程将如何进行吧！</p>
+      </body>
+    </html>
+    ```
+
+3. 刷新你的 “关于” 页面的实时预览。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/3.mdx:89]:
```suggestion
3. 刷新你的 `/about` 页面的实时预览。
```

File: src/content/docs/zh-cn/tutorial/2-pages/4.mdx [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+        font-weight: bold;
+      }
+    </style>
+    ```
+
+
+  4. 再次在浏览器中查看你的关于页面，使用肉眼观察或者浏览器检测工具来验证每个技能列表中的项目现在是否为绿色和加粗。
+
+## 使用你的第一个CSS变量
+Astro 的 `<style>` 标签还可以使用 `define:vars={{ {...} }}` 指令引用 `frontmatter` 中的任何变量。你可以**在代码栏中定义变量，然后在样式标签中将其用作CSS变量使用**。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/4.mdx:85]:
```suggestion
Astro 的 `<style>` 标签还可以使用 `define:vars={ {...} }` 指令引用 `frontmatter` 中的任何变量。你可以**在代码围栏中定义变量，然后在样式标签中将其用作CSS变量使用**。
```

File: src/content/docs/zh-cn/tutorial/2-pages/4.mdx [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+      }
+      .skill {
+        color: green;
+        color: var(--skillColor);
+        font-weight: bold;
+      }
+    </style>
+    ```
+
+3. 在浏览器预览中查看 “关于” 页面。您应该看到技能现在是海军蓝色`navy`，由传递给 define:vars 指令的 skillColor 变量设置。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/2-pages/4.mdx:126]:
```suggestion
3. 在浏览器预览中查看 “关于” 页面。您应该看到技能现在是海军蓝色`navy`，由传递给 `define:vars` 指令的 `skillColor` 变量设置。
```

**Review by liruifengv — CHANGES_REQUESTED** [review]

**Comment by liruifengv** [comment]
I reviewed the first three units, I spotted a few small details.
Tomorrow I will continue.

File: src/content/docs/zh-cn/tutorial/5-astro-api/4.mdx [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+<PreCheck>
+  - 安装 Astro 的包，用于为你的网站创建一个 RSS 订阅源
+  - 创建一个可以通过 RSS 订阅源阅读的订阅源
+</PreCheck>
+
+## 安装 Astro 的 RSS 包
+
+Astro 提供了一个可自定义包，用于快速为你的网站添加一个 RSS 订阅源。
+
+这个官方包会生成一个非 HTML 文档，其中包含有关你的所有博客文章的信息，可以被 Feedly、The Old Reader 等**订阅源阅读器（Feed Reader）**读取。每次重新构建你的网站时，该文档都会更新。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/5-astro-api/4.mdx:22]:
```suggestion
这个官方包会生成一个非 HTML 文档，其中包含有关你的所有博客文章的信息，可以被 Feedly、The Old Reader 等 **订阅源阅读器（Feed Reader）** 读取。每次重新构建你的网站时，该文档都会更新。
```
There is a rendering error here

File: src/content/docs/zh-cn/tutorial/6-islands/1.mdx [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+  <PreactBanner client:load />
+  <SvelteCounter />
+  <SvelteCounter client:visible />
+</BaseLayout>
+```
+
+1. 五个组件中哪些将成为**被 Hydrate 的**岛屿，将其 JavaScript 发送到客户端？
+
+    <p>
+      <Spoiler>`<PreactBanner client:load />` and `<SvelteCounter client:visible />` will be hydrated islands.</Spoiler>
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/6-islands/1.mdx:141]: Translation is missing here.

**Review by liruifengv — COMMENTED** [review]
A little bit of advice. I've reviewed it in its entirety and it's a great tutorial in Chinese.

File: src/content/docs/zh-cn/tutorial/5-astro-api/4.mdx (hunk 15 above)
Comment (liruifengv) [reply src/content/docs/zh-cn/tutorial/5-astro-api/4.mdx:22]: 这里似乎还有问题没解决

File: src/content/docs/zh-cn/tutorial/6-islands/1.mdx (hunk 16 above)
Comment (liruifengv) [reply src/content/docs/zh-cn/tutorial/6-islands/1.mdx:141]: 这里也翻译一下吧

**Review by liruifengv — APPROVED** [review]
LGTM. Amazing work!

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks folks for this fantastic work, let's get this tutorial on the site! 🥳 🎉

**Pull request merged**